### PR TITLE
Bizzclick: host parameter update

### LINKED ISSUE #2912: Port PR from PBS-Go: Bizzclick: adapter update, add new host param
A PR was merged over on PBS-Go

- [https://github.com/prebid/prebid-server/pull/3347](https://github.com/prebid/prebid-server/pull/3347)
- timestamp: 2024-01-18T05:26:38Z

### PR DESCRIPTION
**Pull request by VeryExtraordinaryUsername**
closes #2912

### CODE REVIEW

File: src/main/java/org/prebid/server/bidder/bizzclick/BizzclickBidder.java [hunk 1]
```diff
@@ -100,7 +103,10 @@ private static MultiMap headers(Device device) {
     }
 
     private String buildEndpointUrl(ExtImpBizzclick ext) {
-        return endpointUrl.replace(URL_SOURCE_ID_MACRO, HttpUtil.encodeUrl(ext.getPlacementId()))
+        final String host = StringUtils.isEmpty(ext.getHost()) ? DEFAULT_HOST : ext.getHost();
+        return endpointUrl
+                .replace(URL_HOST_MACRO, HttpUtil.encodeUrl(host))
+                .replace(URL_SOURCE_ID_MACRO, HttpUtil.encodeUrl(ext.getPlacementId()))
```
[the remark this answers is not in the document]
Comment (VeryExtraordinaryUsername) [reply src/main/java/org/prebid/server/bidder/bizzclick/BizzclickBidder.java]: Done. Thx.

File: src/main/java/org/prebid/server/bidder/bizzclick/BizzclickBidder.java [hunk 2]
```diff
@@ -100,7 +103,11 @@ private static MultiMap headers(Device device) {
     }
 
     private String buildEndpointUrl(ExtImpBizzclick ext) {
-        return endpointUrl.replace(URL_SOURCE_ID_MACRO, HttpUtil.encodeUrl(ext.getPlacementId()))
+        final String host = StringUtils.isEmpty(ext.getHost()) ? DEFAULT_HOST : ext.getHost();
+        final String sourceId = StringUtils.isEmpty(ext.getSourceId()) ? ext.getPlacementId() : ext.getSourceId();
```
Comment (marki1an) [inline src/main/java/org/prebid/server/bidder/bizzclick/BizzclickBidder.java:107]: Would be great to have tests for this logic.

**Review by marki1an — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/bidder/bizzclick/BizzclickBidder.java (hunk 2 above)
Comment (VeryExtraordinaryUsername) [reply src/main/java/org/prebid/server/bidder/bizzclick/BizzclickBidder.java:107]: Done.

**Review by marki1an — APPROVED** [review]

File: src/test/resources/org/prebid/server/it/openrtb2/bizzclick/test-auction-bizzclick-request.json [hunk 3]
```diff
@@ -9,6 +9,7 @@
       },
       "ext": {
         "bizzclick": {
+          "host": "host",
           "accountId": "accountId",
           "placementId": "placementId"
```
Comment (AntoxaAntoxic) [inline src/test/resources/org/prebid/server/it/openrtb2/bizzclick/test-auction-bizzclick-request.json:14]: I suggest tweaking the BizzclickTest and the configured endpoint url accordingly to check whether these macros are really resolved

**Review by CTMBNara — APPROVED** [review]

**Review by SerhiiNahornyi — APPROVED** [review]

**Pull request merged**